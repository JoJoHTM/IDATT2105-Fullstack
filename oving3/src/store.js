import { reactive, computed, ref } from 'vue';

const state = reactive({
    feedbackData: {
        name: '',
        email: '',
        feedback: ''
    },
    calculatorData: {
        currentValue: "0",
        totalValue: "0",
        operations: []
    }
});

const disableSubmit = computed(() => {
    return state.feedbackData.name === '' || state.feedbackData.email === '' || state.feedbackData.feedback === '';
});


export default {
    state,
    disableSubmit
};