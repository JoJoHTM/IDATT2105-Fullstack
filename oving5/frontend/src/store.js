import { reactive, computed, ref } from 'vue';

const state = reactive({
    feedbackData: {
        name: '',
        email: '',
        feedback: ''
    },
    calculatorData: {
        currentValue: "0",
        totalValue: "0",
        operations: []
    },
    loginData: {
        username: '',
        password: '',
        loggedIn: false,
        loginID: 0
    },
    registerData: {
        username: '',
        password: ''
    }
});

const disableSubmit = computed(() => {
    return state.feedbackData.name === '' || state.feedbackData.email === '' || state.feedbackData.feedback === '';
});


export default {
    state,
    disableSubmit
};