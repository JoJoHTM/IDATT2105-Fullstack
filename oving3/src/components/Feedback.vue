
<script setup>
import { ref } from 'vue';
import RouterButtons from './RouterButtons.vue';
import store from '../store.js';
import json from '../assets/formSubmission.json';

const { state, disableSubmit } = store;
const submitted = ref(false);


</script>

<template>
    <h1 style="text-align: center;">Tilbakemelding</h1>
    <RouterButtons />
    <Transition>
        <div v-if="submitted" style="text-align: center;">
            <h2>Takk for tilbakemeldingen!</h2>
        </div>
    </Transition>
    <form @submit="submitted = true" v-if="!submitted">
        <label for="name">Navn:</label>
        <input type="text" placeholder="Navn" name="name" id="name" v-model="state.feedbackData.name" required></input>
        <label for="email">E-post:</label>
        <input type="email" placeholder="Email" id="email" name="email" v-model="state.feedbackData.email" required></input>
        <label for="feedback" id="feedbacktekst">Tilbakemelding:</label>
        <textarea placeholder="Skriv din tilbakemelding her" name="feedback" id="feedback" v-model="state.feedbackData.feedback" required></textarea>
        <input type="submit" value="Send" id="submit" :disabled="disableSubmit"></input>
    </form>
</template>

<style scoped>
form {
    display: grid;
    grid-template-areas: 
        'namelabel name name'
        'emaillabel email email'
        'feedbacklabel feedback feedback'
        'send send send';

    width: 30vw;
    height: 60vh;
    min-width: 300px;
    margin: 0 auto;
    padding: 20px;
    border: 1px solid black;
    border-radius: 5px;
}

label {
    font-size: 19px;
}

#name:focus,
#email:focus {
    outline: none;
    border-bottom: 2px solid black;
}

#name {
    grid-area: name;
    height: fit-content;
    border: none;
    border-bottom: 2px solid gray;
}

#email {
    grid-area: email;
    height: fit-content;
    border: none;
    border-bottom: 2px solid gray;
}

#feedback {
    grid-area: feedback;
    resize: none;
    border: 2px solid gray;
}

#submit {
    grid-area: send;
    width: 100px;
    height: 30px;
    border-radius: 15px;
    justify-self: center;
    align-self: center;
}
</style>