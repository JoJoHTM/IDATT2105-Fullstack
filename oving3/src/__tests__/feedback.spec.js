import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';  
import Feedback from '../components/Feedback.vue';
import { createRouter, createWebHistory } from 'vue-router';
import store from '../store.js';

const { state } = store;

const router = createRouter({
    history: createWebHistory(),
    routes: []
});

describe('Feedback', () => {
    it('should render the feedback form', () => {
        const wrapper = mount(Feedback, {
            global: {
                plugins: [router, store]
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have a submit button', () => {
        const wrapper = mount(Feedback, {
            global: {
                plugins: [router, store]
            }
        });
        const button = wrapper.find('input[type="submit"]');
        expect(button.exists()).toBe(true);
    });

    it('should have an input field for feedback', () => {
        const wrapper = mount(Feedback, {
            global: {
                plugins: [router, store]
            }
        });
        const input = wrapper.find('textarea[id="feedback"]');
        expect(input.exists()).toBe(true);
    });

    it('should update the store when feedback is submitted', async () => {
        const wrapper = mount(Feedback, {
            global: {
                plugins: [router, store]
            }
        });
        const input = wrapper.find('textarea[id="feedback"]');
        input.setValue('Great job!');
        await wrapper.find('form').trigger('submit.prevent');
        expect(state.feedbackData.feedback).toBe('Great job!');
    });
});