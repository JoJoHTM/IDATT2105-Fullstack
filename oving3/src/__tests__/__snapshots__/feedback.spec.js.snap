// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Feedback > should render the feedback form 1`] = `
"<h1 data-v-a4640159="" style="text-align: center;">Tilbakemelding</h1>
<div data-v-7101c98d="" data-v-a4640159=""><a data-v-7101c98d="" href="/" class="" style="border-right: 2px solid orange; margin-bottom: 20px;"><button data-v-7101c98d="">Kalkulator</button></a><a data-v-7101c98d="" href="/feedback" class=""><button data-v-7101c98d="">Tilbakemelding</button></a></div>
<dialog data-v-a4640159="" id="popup">
  <h2 data-v-a4640159=""></h2><button data-v-a4640159="" id="closeButton">X</button>
</dialog>
<form data-v-a4640159=""><label data-v-a4640159="" for="name">Navn:</label><input data-v-a4640159="" type="text" placeholder="Navn" id="name" required=""><label data-v-a4640159="" for="email">E-post:</label><input data-v-a4640159="" type="email" placeholder="Email" id="email" required=""><label data-v-a4640159="" for="feedback" id="feedbacktekst">Tilbakemelding:</label><textarea data-v-a4640159="" placeholder="Skriv din tilbakemelding her" id="feedback" required=""></textarea><input data-v-a4640159="" type="submit" id="submit" disabled="" value="Send"></form>"
`;
